pkg/backend/local/progress.go

```diff
@@ -774,9 +775,10 @@ func (display *ProgressDisplay) getRowForURN(urn resource.URN, metadata *engine.
 	// If this is the first time we're seeing an event for the stack resource, check to see if we've already
 	// recorded root events that we want to reassociate with this URN.
 	if isRootURN(urn) {
+		display.stackUrn = urn
+
 		if row, has = display.eventUrnToResourceRow[""]; has {
 			row.SetStep(step)
-			display.stackUrn = urn
```

Comment:
this was bug number one.  we were only setting the display's "stack urn" *if* we heard about the stack *and* we had previously heard about a "global" message.  Now we universally set the display's stackURN 4 lines above.

This is what was causing us to not print outputs.  The stored stackURN was staying ""  which meant this code didn't run:

```ts
	// If we're not previewing, and we get stack outputs, then display them at the end.
	if !display.isPreview {
		if display.stackUrn != "" {
			stackStep := display.eventUrnToResourceRow[display.stackUrn].Step()
```